src/selector.js

```diff
@@ -229,12 +229,13 @@ function find( selector, context, results, seed ) {
 					newContext = rsibling.test( selector ) && testContext( context.parentNode ) ||
 						context;
 
-					// We can use :scope instead of the ID hack if the browser
-					// supports it & if we're not changing the context.
-					if ( newContext !== context || !support.scope ) {
+					// Outside of IE, if we're not changing the context we can
+					// use :scope instead of an ID.
+					if ( newContext !== context || isIE ) {
 
-						// Capture the context ID, setting it first if necessary
-						if ( ( nid = context.getAttribute( "id" ) ) ) {
+						// Capture the context ID, setting it first if necessary.
+						nid = context.getAttribute( "id" );
+						if ( nid ) {
```

Comment:
I actually like this in the `if` better as it's consitent with the expando setting a few lines below.

Reply:
@timmywil Makes sense, I've just reverted this change.